src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reorder_data.cl

```diff
@@ -137,42 +137,90 @@ KERNEL (reorder_data)(
+        res = TO_OUTPUT_REORDER_TYPE(colorRGBA.s0);
+        FUSED_OPS;
+        output[output_idx_0] = FUSED_OPS_RESULT;
+        res = TO_OUTPUT_REORDER_TYPE(colorRGBA.s1);
+        FUSED_OPS;
+        output[output_idx_1] = FUSED_OPS_RESULT;
+        res = TO_OUTPUT_REORDER_TYPE(colorRGBA.s2);
+        FUSED_OPS;
+        output[output_idx_2] = FUSED_OPS_RESULT;
+    #else
+        output[output_idx_0] = ACTIVATION_FUNC_TYPED(OUTPUT_REORDER, TO_OUTPUT_REORDER_TYPE(colorRGBA.s0), NL_M, NL_N);
+        output[output_idx_1] = ACTIVATION_FUNC_TYPED(OUTPUT_REORDER, TO_OUTPUT_REORDER_TYPE(colorRGBA.s1), NL_M, NL_N);
+        output[output_idx_2] = ACTIVATION_FUNC_TYPED(OUTPUT_REORDER, TO_OUTPUT_REORDER_TYPE(colorRGBA.s2), NL_M, NL_N);
+    #endif
+    #if INPUT0_FEATURE_NUM == 4
+        uint8 ov = RESHAPE_DIMS(INPUT0, OUTPUT, b, 3, v, u, w, z, y, x);
+        uint output_idx = FUNC_CALL(get_output_index)(OPTIONAL_SHAPE_INFO_TENSOR ov.s0, ov.s1, ov.s2, ov.s3, ov.s4, ov.s5, ov.s6, ov.s7);
+        #if HAS_FUSED_OPS
+            res = TO_OUTPUT_REORDER_TYPE(colorRGBA.s3);
+            FUSED_OPS;
+            output[output_idx] = FUSED_OPS_RESULT;
+        #else
+            output[output_idx] = ACTIVATION_FUNC_TYPED(OUTPUT_REORDER, TO_OUTPUT_REORDER_TYPE(colorRGBA.s3), NL_M, NL_N);
+        #endif
+    #endif
 #elif OUTPUT_LAYOUT_IMAGE_2D_RGBA
     IMAGE_WRITE(output, (int2)(x, y), colorRGBA);
 #else
-#if INPUT0_IS_FP && !OUTPUT_IS_FP
-#if CONVERT_TRUNCATE
-    output[output_idx] = ACTIVATION_TYPED(OUTPUT_REORDER, TO_OUTPUT_REORDER_TYPE(convert_long(res)), ACTIVATION_PARAMS_TYPED);
-#else
-    output[output_idx] = ACTIVATION_TYPED(OUTPUT_REORDER, TO_OUTPUT_REORDER_TYPE_SAT(res), ACTIVATION_PARAMS_TYPED);
-#endif
-#else
-    output[output_idx] = ACTIVATION_TYPED(OUTPUT_REORDER, TO_OUTPUT_REORDER_TYPE(res), ACTIVATION_PARAMS_TYPED);
-#endif
+    #if INPUT0_IS_FP && !OUTPUT_IS_FP
+        #if CONVERT_TRUNCATE
+            #if HAS_FUSED_OPS
+                res = TO_OUTPUT_REORDER_TYPE(convert_long(res));
+                FUSED_OPS;
+                output[output_idx] = FUSED_OPS_RESULT;
+            #else
+                output[output_idx] = ACTIVATION_TYPED(OUTPUT_REORDER, TO_OUTPUT_REORDER_TYPE(convert_long(res)), ACTIVATION_PARAMS_TYPED);
+            #endif
+        #else
+            #if HAS_FUSED_OPS
+                res = TO_OUTPUT_REORDER_TYPE_SAT(res);
+                FUSED_OPS;
+                output[output_idx] = FUSED_OPS_RESULT;
+            #else
+                output[output_idx] = ACTIVATION_TYPED(OUTPUT_REORDER, TO_OUTPUT_REORDER_TYPE_SAT(res), ACTIVATION_PARAMS_TYPED);
+            #endif
+        #endif
+    #else
+        #if HAS_FUSED_OPS
+            res = TO_OUTPUT_REORDER_TYPE(res);
+            FUSED_OPS;
+            output[output_idx] = FUSED_OPS_RESULT;
+        #else
+            output[output_idx] = ACTIVATION_TYPED(OUTPUT_REORDER, TO_OUTPUT_REORDER_TYPE(res), ACTIVATION_PARAMS_TYPED);
+        #endif
+    #endif
```

Comment:
Now this ifdef seems way to complicated than its purpose. What about having additional macro like below? It will simplify the code.
```
#if INPUT0_IS_FP && !OUTPUT_IS_FP
#if CONVERT_TRUNCATE
    #define __TO_OUTPUT_REORDER_TYPE(res) TO_OUTPUT_REORDER_TYPE(convert_long(res))
#else
    #define __TO_OUTPUT_REORDER_TYPE(res) TO_OUTPUT_REORDER_TYPE_SAT(res)
#endif
#else
    #define __TO_OUTPUT_REORDER_TYPE(res) TO_OUTPUT_REORDER_TYPE(res)
#endif

        #if HAS_FUSED_OPS
            res = __TO_OUTPUT_REORDER_TYPE(res);
            FUSED_OPS;
            output[output_idx] = FUSED_OPS_RESULT;
        #else
            output[output_idx] = ACTIVATION_TYPED(OUTPUT_REORDER, __TO_OUTPUT_REORDER_TYPE(res), ACTIVATION_PARAMS_TYPED);
        #endif

#undef __TO_OUTPUT_REORDER

```

Reply:
Your suggestion made the kernel code definitely more readable. Thanks!